### DISCUSSION TITLE
How to customize validation error msg for unresolved variable?

### DISCUSSION THREAD

**Discussion by durianwaffle** [eclipse-langium/langium#1692] [Q&A] [answered]
Hi Langium community, I am implementing validator for a DSL.

I am wondering how to customize validation error msg for unresolved variable?

Currently, for undeclared variables, the tooltip shows:
![image](https://github.com/user-attachments/assets/dd3e4e6e-80de-4bde-95b2-7910771f8b09)
![image](https://github.com/user-attachments/assets/1edebbac-886e-4d5a-af66-b7ff565f2a7e)

**Comment by msujew** [comment, marked as answer]
Hey @durianwaffle,

You can override the default linker for that, see:

https://github.com/eclipse-langium/langium/blob/335f9a99b1568829cd1c0f5d3bbe33ca754daf90/packages/langium/src/references/linker.ts#L233-L246

**Reply by durianwaffle** [reply]
Thank you @msujew
